### PR TITLE
Add max_surge value to node pool

### PR DESCRIPTION
**Pull request by katiekeel**

### CODE REVIEW

**Review by sudermanjr — CHANGES_REQUESTED** [review]
You'll have to add this variable definition to the variables.tf, along with a description and preferably a default in order to be backward compatible

File: aks_node_pool/variables.tf [hunk 1]
```diff
@@ -54,6 +54,12 @@ variable "max_pods" {
   default     = 200
 }
 
+variable "max_surge" {
+  type        = string
```
Comment (katiekeel) [inline aks_node_pool/variables.tf:58]: As best I can tell, [this should be a string](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool#max_surge). I don't see anything in the Terraform types about percentages being their own thing.

**Comment by katiekeel** [comment]
Ok - here's what the plan output says. It looks like the `max_surge` setting just... didn't exist before? Also, unclear whether its parent `upgrade_settings` block applies to node images upgrades, or just K8s version upgrades.

Screen Shot 2021-08-24 at 11 53 26 AM

**Comment by katiekeel** [comment]
I checked the docs [here](https://docs.microsoft.com/en-us/azure/aks/upgrade-cluster#customize-node-surge-upgrade) and they recommend the `max_surge` be capped at `33%` for prod environments, so I went a bit more conservative and set it to `25%`.

Also, it does explicitly say that `max_surge` [applies to node image upgrades.](https://docs.microsoft.com/en-us/azure/aks/node-image-upgrade#upgrade-node-images-with-node-surge)

**Review by lucasreed — APPROVED** [review]

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**